autoload/ctrlp/funky/ft/sql.vim

```diff
@@ -0,0 +1,21 @@
+" Language: SQL
+" Author: somini
+" License: The MIT License
+
+let s:pattern = '\V\c\<function\>\W\+\w\+(\_.\{-})'
```

Comment:
I tested and Funky showed nothing.
You need to escape parentheses to group the pattern with no very magic (\V).

```sql
CREATE OR REPLACE FUNCTION RIVUS.PRODUCT
(
	P_NUM1 IN NUMBER,
	P_NUM2 IN NUMBER
)
RETURN NUMBER
IS
	vSum	NUMBER(2) DEFAULT  0;
--	vSum	NUMBER(2) :=  0;
BEGIN
	FOR i IN 1..P_NUM2
	LOOP
		vSum := vSum + P_NUM1;
	END LOOP;
	RETURN vSum ;
END;
```